sphinx/ext/inheritance_diagram.py

```diff
@@ -297,9 +300,15 @@ def generate_dot(self, name: str, urls: Dict[str, str] = {},
         for name, fullname, bases, tooltip in sorted(self.class_info):
             # Write the node
             this_node_attrs = n_attrs.copy()
-            if fullname in urls:
-                this_node_attrs['URL'] = '"%s"' % urls[fullname]
-                this_node_attrs['target'] = '"_top"'
+
+            # Fuzzy matching in case the reference is at a different level of the namespace
+            fullname_ends = itemgetter(0, -1)(fullname.split('.'))
+            for refname, url in urls.items():
+                if itemgetter(0, -1)(refname.split('.')) == fullname_ends:
+                    this_node_attrs['URL'] = '"%s"' % url
+                    this_node_attrs['target'] = '"_top"'
+                    break
```

Comment:
I don't know what you're trying to achieve here. Please make the logic more explicit.

Reply:
Moot (explained below)